appium/sample-scripts/python/README.md

```diff
@@ -8,102 +10,183 @@ You'll find here all steps you need to start running your mobile tests
 against real devices in Testdroid Cloud. Before continuing with running with
 these scripts you should register with [Testdroid service](https://cloud.testdroid.com/).
 
-For more detailed guides on Appium please refer to their excellent
+For more detailed guides on Appium please refer to their
 [documentation
 online](http://appium.io/slate/en/master/?python#about-appium).
 
-## Upload Your App To Cloud
+# Uploading Your App To Cloud
+
+Uploading an app is easiest using the `upload.py` script and using
+apiKey identification. The apiKey is found under "My Account" in
+Testdroid Cloud. For the upload to be successful the full path to the
+app (apk or ipa) needs to be provided.
+
+If no app is provided on command line, then the Bitbar Sample Android app is uploaded.
+
+```bash
+$ python upload.py -h
+usage: upload.py [-h] [-k APIKEY] [-a APP_PATH] [-u URL]
+
+Upload a mobile app to Testdroid Cloud and get a handle to it
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -k APIKEY, --apikey APIKEY
+                        User's apiKey to identify to cloud, or set environment
+                        variable TESTDROID_APIKEY
+  -a APP_PATH, --app_path APP_PATH
+                        Path to app to upload or set environment variable
+                        TESTDROID_APP_PATH. Current value is:
+                        '../../../apps/builds/Testdroid.apk'
+  -u URL, --url URL     Testdroid Cloud url to upload app or set environment
+                        variable TESTDROID_UPLOAD_URL. Current value is:
+                        'http://appium.testdroid.com/upload'
+```
 
-Before testing can start you need to upload you mobile application to
-the cloud. To do so open and configure the upload.py script. You will
-need to add your apiKey value that you can get from under "My Account"
-from Testdroid cloud. Also you'll need to set the full path to your
-mobile app. This can be an Android or iOS application.
+The below example shows how to upload a hybrid Android demo app to Testdroid Cloud.
 
-```sh
-$ python upload.py 
-Filename to use in testdroid capabilities in your test: 719f52c4-2c34-4c25-b90b-08884f049d3a/Testdroid.apk
+```bash
+$ python upload.py -k xg8x...YXto -u ../../../apps/builds/Testdroid.apk
+Filename to use in testdroid capabilities in your test: 719f52c4-43c2-4c25-b91b-08884f049d3a/Testdroid.apk
 ```
 
-From the response message you need to store the application's file
-name in the cloud. In this example upload it is
-'719f52c4-2c34-4c25-b90b-08884f049d3a/Testdroid.apk'.
+The response message provides the application's cloud file
+name that is needed to start testing against the app. From the above example the path to store is:
```

Comment:
"the path to store is:" seems confusing choice of word for me.

How about "the path to use is:" or "the path to use with testdroid_app is:"